src/InMemoryProvider.ts

```diff
@@ -24,11 +24,34 @@ export interface StoreData {
     schema: StoreSchema;
 }
 
+let asyncCallbacks: Set<() => void> = new Set();
+
+/**
+ * This function will defer callback of the specified callback lambda until the next JS tick, simulating standard A+ promise behavior
+ */
+function asyncCallback(callback: () => void): void {
+    asyncCallbacks.add(callback);
+
+    if (asyncCallbacks.size === 1) {
+        setImmediate(resolveAsyncCallbacks);
```

Comment:
setImmediate isnt recommended to be used in production environments. Based on CanIUse this should only really exist on IE https://caniuse.com/?search=Setimmediate
For support/compatility settimeout should probably use used instead.

Reply:
The UTs did work ... I think the API is being polyfilled in somehow 
![image](https://user-images.githubusercontent.com/942002/100487286-9cf1a880-30bc-11eb-8687-1f56a6b6ac01.png)

Reply:
So the polyfill is coming from 
https://github.com/zloirock/core-js/blob/v3.8.0/packages/core-js/internals/task.js
Which is imported by 
https://github.com/zloirock/core-js/blob/v3.8.0/packages/core-js/modules/es.promise.js
which is imported by 
 'core-js/features/promise'
which is imported by 
src\Promise.extensions.ts

I can create a follow up PR to explicitly import this setImmediate polyfill.. would that address the concerns raised here?